x-pack/plugins/security_solution/cypress/integration/detection_rules/bulk_edit_rules.spec.ts

```diff
@@ -90,180 +130,358 @@ describe('Detection rules, bulk edit', () => {
+
+      // check if only pre-populated tags exist in the tags filter
+      checkTagsInTagsFilter(prePopulatedTags);
+
+      cy.get(EUI_FILTER_SELECT_ITEM)
+        .should('have.length', prePopulatedTags.length)
+        .each(($el, index) => {
+          cy.wrap($el).should('have.text', prePopulatedTags[index]);
+        });
+
+      selectNumberOfRules(expectedNumberOfCustomRulesToBeEdited);
+
+      // open add tags form and add 2 new tags
+      openBulkEditAddTagsForm();
+      typeTags(tagsToBeAdded);
+      submitBulkEditForm();
+      waitForBulkEditActionToFinish({ rulesCount: expectedNumberOfCustomRulesToBeEdited });
+
+      // check if all rules have been updated with new tags
+      testAllTagsBadges(resultingTags);
+
+      // check that new tags were added to tags filter
+      // tags in tags filter sorted alphabetically
+      const resultingTagsInFilter = [...resultingTags].sort();
+      checkTagsInTagsFilter(resultingTagsInFilter);
+    });
+
+    it('Overwrite tags in custom rules', () => {
+      const tagsToOverwrite = ['overwrite-tag-1'];
+
+      // check if only pre-populated tags exist in the tags filter
+      checkTagsInTagsFilter(prePopulatedTags);
+
+      selectNumberOfRules(expectedNumberOfCustomRulesToBeEdited);
+
+      // open add tags form, check overwrite tags and warning message, type tags
+      openBulkEditAddTagsForm();
+      checkOverwriteTagsCheckbox();
+
+      cy.get(RULES_BULK_EDIT_TAGS_WARNING).should(
+        'have.text',
+        `You’re about to overwrite tags for ${expectedNumberOfCustomRulesToBeEdited} selected rules, press Save to apply changes.`
+      );
+
+      typeTags(tagsToOverwrite);
+      submitBulkEditForm();
+      waitForBulkEditActionToFinish({ rulesCount: expectedNumberOfCustomRulesToBeEdited });
+
+      // check if all rules have been updated with new tags
+      testAllTagsBadges(tagsToOverwrite);
+
+      // check that only new tags are in the tag filter
+      checkTagsInTagsFilter(tagsToOverwrite);
+    });
+
+    it('Delete tags from custom rules', () => {
+      const tagsToDelete = prePopulatedTags.slice(0, 1);
+      const resultingTags = prePopulatedTags.slice(1);
+
+      // check if only pre-populated tags exist in the tags filter
+      checkTagsInTagsFilter(prePopulatedTags);
+
+      selectNumberOfRules(expectedNumberOfCustomRulesToBeEdited);
+
+      // open add tags form, check overwrite tags, type tags
+      openBulkEditDeleteTagsForm();
+      typeTags(tagsToDelete);
+      submitBulkEditForm();
+      waitForBulkEditActionToFinish({ rulesCount: expectedNumberOfCustomRulesToBeEdited });
+
+      // check tags has been removed from all rules
+      testAllTagsBadges(resultingTags);
 
-    // check if rule has been updated
-    goToTheRuleDetailsOf(RULE_NAME);
-    hasIndexPatterns(OVERWRITE_INDEX_PATTERNS.join(''));
+      // check that tags were removed from the tag filter
+      checkTagsInTagsFilter(resultingTags);
```

Comment:
There's an edge case here in which the deleted tag doesn't get removed from the TagsFilterDropdown if the deleted tag was selected in the filter before the delete takes place.
I detailed it in: https://docs.google.com/document/d/116x7ITTTJQ6cTiwaGK831_f6Ox7XB3qyLiHxC3Cmf8w/edit?disco=AAAAevQEkp8

I think fixing that bug is out of scope of this testing PR, but wanted to leave the note here. Will create a follow up issue for it.

Reply:
thanks for catching this 👍 . Dedicated issue for it sounds perfect!